### PR TITLE
[fix](grouping sets) fix grouping sets have multiple empty sets.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

in this https://github.com/apache/doris/pull/32112, handling empty sets (empty expression cases) has been addressed. However, multiple empty sets in grouping sets have different grouping IDs

https://dbfiddle.uk/VYe7V2Mn

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

File: be/src/pipeline/exec/repeat_operator.cpp [hunk 1]
```diff
@@ -228,8 +235,16 @@ Status RepeatOperatorX::pull(doris::RuntimeState* state, vectorized::Block* outp
...
-        DCHECK(!_intermediate_block || (_intermediate_block && _intermediate_block->rows() == 0));
-        output_block->swap(_child_block);
+        auto m_block = vectorized::VectorizedUtils::build_mutable_mem_reuse_block(output_block,
+                                                                                  _output_slots);
+        auto rows = _child_block.rows();
+        auto& columns = m_block.mutable_columns();
+
+        for (int repeat_id_idx = 0; repeat_id_idx < _repeat_id_list.size(); repeat_id_idx++) {
+            std::size_t cur_col = 0;
+            RETURN_IF_ERROR(
```
Comment (HappenLee) [inline be/src/pipeline/exec/repeat_operator.cpp:245]: why call ```local_state.add_grouping_id_column(rows, 0, columns, repeat_id_idx));```

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**